Kubernetes/legos/k8s_remove_pod_from_deployment/k8s_remove_pod_from_deployment.py

```diff
@@ -0,0 +1,92 @@
+##
+# Copyright (c) 2023 unSkript, Inc
+# All rights reserved.
+##
+import pprint
+
+from kubernetes import client
+from typing import Optional 
+from pydantic import BaseModel, Field
+
+class InputSchema(BaseModel):
+    pod_name: str = Field(
+        title="Pod Name",
+        description="K8S Pod Name"
+    )
+    namespace: str = Field(
+        title="Namespace",
+        description="K8S Namespace where the POD exists"
+    )
+    deployment_label: Optional[str] = Field(
```

Comment:
We never talked about asking for this? Why cant we figure out the deployment label from the pod details? I found some code around it
```
# the pod owner
from kubernetes import client
core_api = client.CoreV1Api()
apis_api = client.AppsV1Api()
pod = core_api.read_namespaced_pod(name=name, namespace=namespace)
owner_references = pod.metadata.owner_references
if isinstance(owner_references, list):
    owner_name = owner_references[0].name
    owner_kind = owner_references[0].kind
    # owner  StatefulSet
    if owner_kind == 'ReplicaSet':
        replica_set = apis_api.read_namespaced_replica_set(name=owner_name, namespace=namespace)
        owner_references2 = replica_set.metadata.owner_references
        if isinstance(owner_references2, list):
            print(owner_references2[0].name)
        else:
            print(owner_name)
    # owner Deployment
    else:
        print(owner_name)
print(name)
```

Reply:
Done